src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -16,18 +16,257 @@
+        /// <item>Partial.</item>
+        /// </list>
         /// </summary>
-        private SourceOrdinaryMethodSymbol _otherPartOfPartial;
+        private sealed class SourceOrdinaryMethodSymbolUncommon : SourceOrdinaryMethodSymbol
+        {
+            private readonly TypeSymbol _explicitInterfaceType;
+
+            private readonly TypeParameterInfo _typeParameterInfo;
+
+            /// <summary>
+            /// If this symbol represents a partial method definition or implementation part, its other part (if any).
+            /// This should be set, if at all, before this symbol appears among the members of its owner.  
+            /// The implementation part is not listed among the "members" of the enclosing type.
+            /// </summary>
+            private SourceOrdinaryMethodSymbol _otherPartOfPartial;
+
+            public SourceOrdinaryMethodSymbolUncommon(
+                NamedTypeSymbol containingType,
+                TypeSymbol explicitInterfaceType,
+                string name,
+                Location location,
+                MethodDeclarationSyntax syntax,
+                MethodKind methodKind,
+                bool isNullableAnalysisEnabled,
+                BindingDiagnosticBag diagnostics)
+                : base(containingType, name, location, syntax, methodKind, isNullableAnalysisEnabled, diagnostics)
+            {
+                _explicitInterfaceType = explicitInterfaceType;
+                _typeParameterInfo = new TypeParameterInfo { LazyTypeParameters = MakeTypeParameters(syntax, diagnostics) };
+            }
+
+            protected sealed override TypeSymbol ExplicitInterfaceType => _explicitInterfaceType;
+            internal sealed override SourceOrdinaryMethodSymbol OtherPartOfPartial => _otherPartOfPartial;
+
+            internal static void InitializePartialMethodParts(SourceOrdinaryMethodSymbolUncommon definition, SourceOrdinaryMethodSymbolUncommon implementation)
+            {
+                Debug.Assert(definition.IsPartialDefinition);
+                Debug.Assert(implementation.IsPartialImplementation);
+
+                Debug.Assert(definition._otherPartOfPartial is null || definition._otherPartOfPartial == implementation);
+                Debug.Assert(implementation._otherPartOfPartial is null || implementation._otherPartOfPartial == definition);
+
+                definition._otherPartOfPartial = implementation;
+                implementation._otherPartOfPartial = definition;
+            }
+
+            protected sealed override MethodSymbol FindExplicitlyImplementedMethod(BindingDiagnosticBag diagnostics)
+            {
+                var syntax = GetSyntax();
+                return this.FindExplicitlyImplementedMethod(isOperator: false, _explicitInterfaceType, syntax.Identifier.ValueText, syntax.ExplicitInterfaceSpecifier, diagnostics);
+            }
+
+            public sealed override ImmutableArray<TypeParameterSymbol> TypeParameters
```

Comment:
>public sealed override ImmutableArray<TypeParameterSymbol> TypeParameters [](http://example.com/codeflow?start=12&length=73)

Please place this method at its original position to red uce the diff noise. The only diff I expect to see there is the addition of the `sealed` modifier.

Reply:
Scratch that, I see that the original position belongs to a different class.